### PR TITLE
Rewrite / reorganize the troubleshooting documentation

### PR DESCRIPTION
**Pull request by rchl**
- common problems moved to the beginning
- common problems rewritten to be more concise, remove repetition
- in "Completions not shown after certain keywords" removed note about python completion inhibiting issue as that is fixed
- other small changes that are hard to summarize

### CODE REVIEW

**Comment by rchl** [comment]
I'll move back stuff to the previous places to hopefully make diff easier to read.
Will revert to current locations after review is done.

File: docs/src/troubleshooting.md [hunk 1]
```diff
@@ -15,13 +15,11 @@ If the server is crashing on startup, try running `LSP: Troubleshoot server` fro
 
 ## Updating the PATH used by LSP servers
 
-You can confirm that your issue is due to `PATH` being different by starting Sublime Text from the command line so that it inherits your shell's environment.
+Sublime Text might see a different `PATH` from what your shell environment uses and might not be able to find the server binary due to that. You can see what ST thinks your `PATH` is by opening the ST console by clicking on *View > Show Console*, and running `import os; os.environ["PATH"]` in that console.
```
Comment (rchl) [inline docs/src/troubleshooting.md:18]: The old text started with "You can confirm that your issue is due..." even though there is no any issue in this context.

File: docs/src/troubleshooting.md [hunk 2]
```diff
@@ -61,43 +68,29 @@ Another solution could be (at least on Linux) to update the server `PATH` using
 
 ### 1. LSP doesn't start my language server
 
-* Make sure you have a folder added in your Sublime workspace.
-* Make sure the document you are opening lives under that folder.
```
Comment (rchl) [inline docs/src/troubleshooting.md:65]: Removed as this is no longer the case, we pretty much always start the server. In some cases like LSP-typescript we prevent the start but that gives some feedback in the status bar.

File: docs/src/troubleshooting.md [hunk 3]
```diff
@@ -61,43 +68,29 @@ Another solution could be (at least on Linux) to update the server `PATH` using
...
-
-Your client configuration requires two settings to match the document your are editing:
+When language server is started, its name appears on the left side of the status bar. If you expect your server to start for a particular file but it doesn't then:
 
-* Scope (eg. `source.php`): Verify this is correct by running "Show Scope Name" from the developer menu.
-* Syntax (eg. `Packages\PHP\PHP.sublime-syntax`): Verify by running `view.settings().get("syntax")` in the console.
```
Comment (rchl) [inline docs/src/troubleshooting.md:70]: The syntax obviously no longer matters so removed.

File: docs/src/troubleshooting.md [hunk 4]
```diff
@@ -61,43 +68,29 @@ Another solution could be (at least on Linux) to update the server `PATH` using
...
 
 ### 2. LSP cannot find my language server (`No such file or directory: 'xyz'`)
-Assuming that the server is actually installed, and that you can start it from your shell, this issue is likely due to Sublime Text's internal environment not picking up the same `PATH` environment variable as you've configured in your shell.
 
-The exact changes to make can differ depending on what program you want to expose to Sublime Text. The simplest way is to extend the path like so (replacing `/usr/local/bin` with the path of your choice):
-```sh
-export PATH="/usr/local/bin:$PATH"
-```
-
-If, for example, you want to expose a `Node` binary to ST and you have it installed through a version manager like `nvm`, you need to insert its [initialization script](https://github.com/nvm-sh/nvm#install--update-script) in the location specified in [this table](troubleshooting.md#updating-the-path-used-by-lsp-servers)
```
Comment (rchl) [inline docs/src/troubleshooting.md:80]: This part about "nvm" moved to the "Updating the PATH used by LSP servers" section

**Review by predragnikolic — APPROVED** [review]

**Pull request merged**